includes/migrations/class-wc-stripe-subscriptions-repairer-legacy-sepa-tokens.php

```diff
@@ -140,4 +161,170 @@ public function maybe_migrate_before_renewal( $subscription_id ) {
 			$token_updater->maybe_update_subscription_source( $subscription );
 		}
 	}
+
+	/**
+	 * Displays an admin notice to inform the user that the repair is in progress.
+	 *
+	 * This notice is displayed on the Subscriptions list table page and includes information about the progress of the repair.
+	 * What % of the repair is complete, or when the next scheduled action is expected to run.
+	 */
+	public function display_admin_notice() {
+
+		if ( ! class_exists( 'WC_Subscriptions' ) || ! WC_Stripe_Feature_Flags::is_upe_checkout_enabled() ) {
+			return;
+		}
+
+		// Only display this on the subscriptions list table page.
+		if ( ! $this->is_admin_subscriptions_list_table_screen() ) {
+			return;
+		}
+
+		// The notice is only displayed for up to 3 days after disabling the setting.
+		$display_notice = get_transient( $this->display_notice_transient ) === 'yes';
+
+		if ( ! $display_notice ) {
+			return;
+		}
+
+		// Check if there are subscriptions to be migrated.
+		$subscriptions = wc_get_orders(
+			[
+				'return'         => 'ids',
+				'type'           => 'shop_subscription',
+				'status'         => 'any',
+				'posts_per_page' => 1,
+				'payment_method' => WC_Gateway_Stripe_Sepa::ID,
+			]
+		);
+
+		if ( empty( $subscriptions ) ) {
+			delete_transient( $this->display_notice_transient );
+			return;
+		}
+
+		$is_scheduling_jobs = is_numeric( as_next_scheduled_action( $this->scheduled_hook ) );
+		$next_scheduled_job = $this->get_next_scheduled_update_time();
+		$progress           = '';
+
+		if ( $next_scheduled_job ) {
+			// translators: %1$s: <strong> tag, %2$s: </strong> tag, %3$s: human-readable time diff or date.
+			$progress = sprintf( __( '%1$sNext progress update expected: %2$s %3$s', 'woocommerce-gateway-stripe' ), '<strong>', '</strong>', $next_scheduled_job );
```

Comment:
Just a thought after looking at the notice. Could we use the count of subscriptions above to add a bit more context here? i.e.

"Preparing to update 1200 SEPA subscriptions. Expected to start 53 minutes" or something like that?

Reply:
We spoke about this IRL but the answer is no. We only get the counts from the number of Scheduled actions, we don't have a count of Subscriptions with the legacy SEPA payment method set.